chore: migrated institutional ui components to ts

albertolive opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Most of our components are still written in JS; there’s nothing wrong with that, but since MM has been making an effort to move their components to TS and all our new UI pieces are written in TS, we should start moving our UI elements to TS, starting from:

`ui/components/institutional/`

AC:
- At least 50% of our components have been updated to TS along with their unit tests. (start with the smaller, easier components first and then afterwards move to the more complex ones and probably split it into its tickets)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

albertolive committed:
chore: migrated jwt-url-form to ts

albertolive committed:
chore: migrated custody-labels to ts

albertolive committed:
chore: migrated confirm-remove-jwt-modal to ts

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
chore: migrated custody-confirm-link-modal to ts

albertolive committed:
chore: migrated note-to-trader to ts

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
fix: fixed eslint issues

albertolive committed:
chore: migrated wrong-network-notification to ts

albertolive committed:
chore: migrated transaction-failed-modal to ts

albertolive committed:
chore: migrated jwt-dropdown to ts

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
fix: tests

albertolive committed:
chore: migrated interactive-replacement-token-modal to ts

albertolive committed:
chore: migrated interactive-replacement-token-notification to ts

albertolive committed:
fix: fixed storybook in interactive-replacement-token-notification

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
chore: fixed ts issues

albertolive committed:
fix: tests

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
fix: fixed estlin issues

albertolive committed:
fix: eslint issues

albertolive committed:
fix: eslint issue

pedronfigueiredo commented on ui/components/institutional/note-to-trader/note-to-trader.stories.tsx:
```diff
@@ -9,13 +9,13 @@ export default {
 };
 
-export const DefaultStory = (args) => <NoteToTrader {...args} />;
+export const DefaultStory = (args: any) => <NoteToTrader {...args} />;
```
In this case we can define a type instead of any from observing the `args` values above

albertolive replied:
@pedronfigueiredo changes are done

albertolive committed:
fix: removed any from files

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

micaelae commented on ui/components/institutional/wrong-network-notification/wrong-network-notification.tsx:
```diff
@@ -14,12 +14,19 @@ import { useI18nContext } from '../../../hooks/useI18nContext';
 import { Icon, IconName, IconSize, Box, Text } from '../../component-library';
 
-const WrongNetworkNotification = () => {
+type ProviderConfig = {
+  nickname?: string;
+  type: string;
+};
```
Did you consider using the `ProviderConfig` definition from `@metamask/network-controller` instead of defining a new type?

micaelae commented on ui/components/institutional/interactive-replacement-token-notification/interactive-replacement-token-notification.tsx:
```diff
@@ -93,8 +100,16 @@ const InteractiveReplacementTokenNotification = ({ isVisible }) => {
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [address, interactiveReplacementToken.oldRefreshToken, isUnlocked]);
+  }, [
+    address,
+    interactiveReplacementToken.oldRefreshToken,
+    isUnlocked,
+    dispatch,
+    keyring.type,
+    interactiveReplacementToken,
+    mmiActions,
+    address,
```
`address` is listed twice and one can be removed

albertolive replied:
Good catch!

micaelae commented on ui/components/institutional/interactive-replacement-token-modal/interactive-replacement-token-modal.tsx:
```diff
@@ -0,0 +1,156 @@
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+
+type Custodian = {
+  envName: string;
+  iconUrl?: string;
+  displayName: string;
+};
```
Is it possible to use `ICustodianType` from @metamask-institutional instead?

micaelae commented on ui/components/institutional/custody-confirm-link-modal/custody-confirm-link-modal.tsx:
```diff
@@ -28,16 +27,49 @@ import {
+  hideModal: () => void;
+};
+
+type CustodianItem = {
```
same question re: `ICustodianType`

albertolive committed:
fix: improved ts

albertolive committed:
Merge branch 'MMI-5205-mmi-ui-components-to-ts' of https://github.com/MetaMask/metamask-extension into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
feat: added metamask-institutional/types dependency

albertolive commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

albertolive committed:
fix: eslint issue

albertolive committed:
fix: eslint issue

pedronfigueiredo reviewed (changes requested):
I think we should try to remove the remaining `any` declarations, see [the contributor docs](https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#avoid-any) for more info, and follow [a recent thread on this](https://consensys.slack.com/archives/CTQAGKY5V/p1721656680887459).

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive commented:
> I think we should try to remove the remaining `any` declarations, see [the contributor docs](https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#avoid-any) for more info, and follow [a recent thread on this](https://consensys.slack.com/archives/CTQAGKY5V/p1721656680887459).

@pedronfigueiredo I already removed all of the `any` that I introduced. There is only one left, `ui/components/institutional/wrong-network-notification/wrong-network-notification.stories.tsx` but that component is deprecated and removed in this PR https://github.com/MetaMask/metamask-extension/pull/26000

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts

albertolive committed:
Merge branch 'develop' into MMI-5205-mmi-ui-components-to-ts
